pkg/resourcecollector/role.go

```diff
@@ -41,10 +41,27 @@ func (r *ResourceCollector) prepareRoleBindingForApply(
 	if err != nil {
 		return err
 	}
-	rb.Subjects, err = r.updateSubjects(rb.Subjects, namespaceMappings)
+	subjectWithNamespace := make([]rbacv1.Subject, 0)
+	// Create list of rolebinding subjects that has namspace and pass it to updateSubject to
+	// update destination namespace based on the namespace mapping.
+	for _, subject := range rb.Subjects {
+		if len(subject.Namespace) != 0 {
+			subjectWithNamespace = append(subjectWithNamespace, subject)
+		}
+	}
+	subjectWithNamespace, err = r.updateSubjects(subjectWithNamespace, namespaceMappings)
 	if err != nil {
 		return err
 	}
+	//Update the rolebinding subject again with the updated subject value.
+	for sIndex, subject := range subjectWithNamespace {
+		for rbIndex, rbSubject := range rb.Subjects {
+			if rbSubject.Name == subject.Name {
```

Comment:
This won't handle the scenario where the name is same across two different subjects of different kind in the same RoleBinding
```
subjects:
- kind: ServiceAccount
  name: px-backup
- kind: Group
  name: px-backup
```

Reply:
Fixed.